RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -555,11 +555,12 @@ void Event::createTriplets() {
 }
 
 void Event::createTrackCandidates(bool no_pls_dupclean, bool tc_pls_triplets) {
-  if (trackCandidatesInGPU == nullptr) {
-    trackCandidatesInGPU = new TrackCandidates();
-    trackCandidatesBuffers = new TrackCandidatesBuffer<Device>(
-        n_max_nonpixel_track_candidates + n_max_pixel_track_candidates, devAcc, queue);
-    trackCandidatesInGPU->setData(*trackCandidatesBuffers);
+  if (trackCandidatesD_ == nullptr) {
+    trackCandidatesDC_ =
+        new TrackCandidatesDeviceCollection(n_max_nonpixel_track_candidates + n_max_pixel_track_candidates, queue);
+    auto buf = trackCandidatesDC_->buffer();
+    alpaka::memset(queue, buf, 0u);
```

Comment:
Is it worth initializing the entire buffer to zeros? Maybe we can just add a call to `initScalars` at the beginning of `AddpT3asTrackCandidatesInGPU`?

Reply:
In addition to scalars the following were reset to 0 previously: `logicalLayers_buf, lowerModuleIndices_buf, hitIndices_buf, pixelSeedIndex_buf`; this is most of the buffer. So, I opted to reset it all.